### PR TITLE
Add more logging for tasks and referer

### PR DESCRIPTION
**Pull request by nabilschear**
### Description of the Change

Trying to get parity between IMDS logging that we had in baseami with titus.  Just adding a couple of fields to make correlation easier.

### CODE REVIEW

File: metadataserver/server.go [hunk 1]
```diff
@@ -235,6 +235,8 @@ func (ms *MetadataServer) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		"uri":           r.RequestURI,
 		"path":          r.URL.Path,
 		"token-present": len(r.Header.Get("X-aws-ec2-metadata-token")) > 0,
+		"referer":       r.Header.Get("Referer"),
+		"taskid":        ms.titusTaskInstanceID,
```
Comment (sargun) [inline metadataserver/server.go:239]: So, task ID is already present in the logs in the systemd unit name. Do you want duplication?

Comment (nabilschear) [reply metadataserver/server.go:239]: yeah, i saw that.  it's a pain to get out of there in elastic or bpd.  requires tokenizing the unit name and what not.

Comment (sargun) [reply metadataserver/server.go:239]: I know that it's a pain to get from elastic, but it shouldn't be from BDP. Let me know if you need help with that.

**Review by sargun — APPROVED** [review]

**Comment by nabilschear** [comment]
Hold off on merging this.  i'm gonna use this to build some other metadata service changes

**Comment by sargun** [comment]
lgtm.

**Pull request merged**